ssl_cert_tracker/utils.py

```diff
@@ -14,19 +14,20 @@
 
 """
 import logging
-from datetime import datetime
 import xml.dom.minidom
+from dateutil import parser
 
 logging.basicConfig(filename='p_soc_auto.log', level=logging.DEBUG)
 
 def validate(date_text):
     """check if date_text is a valid date  """
     try:
-        if date_text != datetime.strptime(date_text, "%Y-%m-%d").strftime('%Y-%m-%d'):
-            raise ValueError
+        parser.parse(date_text[0:10])
         return True
     except TypeError:
         return False
+    except IndexError:

```

Comment:
the except blocks can be condensed thusly:

```python
except (TypeError, IndexError, ValueError):
    return False
```

but happens if the exception is neither of the ones you are catching? the proper pattern for this kind of processing is

```python
def validate(date_text):
    try:
        parser.parse(date_text[0:10])
        return True
    except:    # pylint:disable=....
        return False